classes/class-wc-connect-nux.php

```diff
@@ -285,12 +319,17 @@ public function show_banner_after_connection() {
 
 			// Did the user just dismiss?
 			if ( isset( $_GET['wcs-nux-notice'] ) && 'dismiss' === $_GET['wcs-nux-notice'] ) {
-				WC_Connect_Options::update_option( self::SUCCESS_BANNER_IS_DISMISSED, true );
+				// No longer need to keep track of whether the before connection banner was displayed.
+				WC_Connect_Options::delete_option( self::SHOULD_SHOW_AFTER_CXN_BANNER );
 				wp_safe_redirect( remove_query_arg( 'wcs-nux-notice' ) );
+				exit;
 			}
 
+			// By going through the connection process, the user has accepted our TOS
+			WC_Connect_Options::update_option( 'tos_accepted', true );
```

Comment:
I don't think we should do that because the JP connection can fail, and that would mean that legally they haven't accepted TOS yet, no?